src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,156 @@
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is 0", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(0);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    describe("getEventFee should correctly apply a reward multiplier", () => {
+      for (const multiplier of [-23, 0, 23]) {
+        const signToZero = multiplier === 0 ? "==" : multiplier > 0 ? ">" : "<";
+        it(`should return a discounted balance fee if the multiplier is ${signToZero} 0`, () => {
+          const amount = BigNumber.from(10);
+          const lastRunningBalance = BigNumber.from(1000);
+          const lastIncentiveBalance = BigNumber.from(1000);
+          const chainId = 1;
+          const flowType = "outflow";
+
+          const originalFee = getEventFee(
+            amount,
+            flowType,
+            lastRunningBalance,
+            lastIncentiveBalance,
+            chainId,
+            defaultConfig
+          );
+
+          const config = new MockUBAConfig();
+          config.setRewardMultiplier(chainId.toString(), toBN(multiplier));
+
+          const modifiedFee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, config);
+          expect(modifiedFee.balancingFee.toString()).toEqual(originalFee.balancingFee.mul(multiplier).toString());
+        });
+      }
+    });
+
+    it("should return a discounted balance fee if the lastIncentiveBalance is positive", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "inflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("20");
+    });
+
+    it("should apply a discount if the incentive balance is below the feesToBringFeePctToZero", () => {
+      const config = new MockUBAConfig();
+      // These values are set such that our feeToBringFeePctToZero is higher than the balancing fee
+      // within the getEventFee function. This is to ensure that the discount is applied for this test
+      // case.
+      const amount = BigNumber.from(1000);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "outflow";
```

Comment:
Can you add an `expect` to log out that the `feeToBringFeePctToZero.gt(lastIncentiveBalance)`? So we don't have a false positive test on some complex logic

Also, why is the `balancingFee` 20? Shouldn't it be discounted?

Reply:
Done - these were from Chase's original truth conditions. I've just tried to port over the code.

Reply:
This shouldn't rely at all on chase's truth conditions we should be reasoning only by looking at the code within `getEventFee`. This test in particular should be testing lines 177-185 here: https://github.com/across-protocol/sdk-v2/blob/master/src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.ts#L184